metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -329,21 +339,67 @@ func GetDefaultDisabledMetaConfig() *kubernetesConfig {
 	}
 }
 
-func ValidatedConfig(base mb.BaseMetricSet) *kubernetesConfig {
-	config := kubernetesConfig{
+func GetValidatedConfig(base mb.BaseMetricSet) (*kubernetesConfig, error) {
+	config, err := GetConfig(base)
+	if err != nil {
+		logp.Err("Error while getting config: %v", err)
+		return nil, err
+	}
+
+	moduleConfig := base.Module().Config()
+	cacheTimeout := PerfMetrics.Timeout
+	config, err = validateConfig(config, moduleConfig, cacheTimeout)
+	if err != nil {
+		logp.Err("Error while validating config: %v", err)
+		return nil, err
+	}
+	return config, nil
+}
+
+func validateConfig(config *kubernetesConfig, moduleConfig mb.ModuleConfig, cacheTimeout time.Duration) (*kubernetesConfig, error) {
+	// NOTE: minimum cache expiration time > period avoid that the cache expires just
+	// before next scraping time. Period * 2 is arbitrary but it a good buffer to avoid
+	// that a late scraping run would affect the metrics (for the absence of entries in the cache)
+	minCacheExpirationTime := moduleConfig.Period * 2
+
+	if config.CacheExpirationTime <= 0 {
+		if cacheTimeout < minCacheExpirationTime {
+			config.CacheExpirationTime = minCacheExpirationTime
+			logp.Debug("setting CacheExpirationTime = minCacheExpirationTime. CacheExpirationTime: %s", config.CacheExpirationTime.String())
```

Comment:
I would use sth more user friendly like `setting cache_expiration_time to minimum default: %s`